src/rust/generators/sysmon-subgraph-generator/src/models/process/create.rs

```diff
@@ -91,3 +85,66 @@ pub fn generate_process_create_subgraph(
 
     Ok(graph)
 }
+#[cfg(test)]
+mod tests {
+    use grapl_graph_descriptions::{graph_description::node_property::Property,
+                                   ImmutableUintProp};
+    use sysmon::*;
+
+    use super::*;
+
+    fn find_node<'a>(
+        graph: &'a GraphDescription,
+        o_p_name: &str,
+        o_p_value: Property,
+    ) -> Option<&'a NodeDescription> {
+        graph.nodes.values().find(|n| {
+            n.properties
+                .iter()
+                .find(|(p_name, p_value)| {
+                    &p_name.as_str() == &o_p_name && p_value.property.clone().unwrap() == o_p_value
+                })
+                .is_some()
+        })
+    }
+
+    #[test]
+    fn process_create() {
+        // Given - A sysmon process creation event
+        // When - We generate a graph from the event
+        // Then - We expect a graph with a parent and child process, and an edge between them
+
+        let event = r#"<Event xmlns='http://schemas.microsoft.com/win/2004/08/events/event'><System><Provider Name='Microsoft-Windows-Sysmon' Guid='{5770385F-C22A-43E0-BF4C-06F5698FFBD9}'/><EventID>1</EventID><Version>5</Version><Level>4</Level><Task>1</Task><Opcode>0</Opcode><Keywords>0x8000000000000000</Keywords><TimeCreated SystemTime='2019-07-24T18:05:14.402156600Z'/><EventRecordID>550</EventRecordID><Correlation/><Execution ProcessID='3324' ThreadID='3220'/><Channel>Microsoft-Windows-Sysmon/Operational</Channel><Computer>DESKTOP-FVSHABR</Computer><Security UserID='S-1-5-18'/></System><EventData><Data Name='RuleName'></Data><Data Name='UtcTime'>2019-07-24 18:05:14.399</Data><Data Name='ProcessGuid'>{87E8D3BD-9DDA-5D38-0000-0010A3941D00}</Data><Data Name='ProcessId'>5752</Data><Data Name='Image'>C:\Windows\System32\cmd.exe</Data><Data Name='FileVersion'>10.0.10240.16384 (th1.150709-1700)</Data><Data Name='Description'>Windows Command Processor</Data><Data Name='Product'>Microsoft� Windows� Operating System</Data><Data Name='Company'>Microsoft Corporation</Data><Data Name='OriginalFileName'>Cmd.Exe</Data><Data Name='CommandLine'>"cmd" /C "msiexec /quiet /i cmd.msi"</Data><Data Name='CurrentDirectory'>C:\Users\grapltest\Downloads\</Data><Data Name='User'>DESKTOP-FVSHABR\grapltest</Data><Data Name='LogonGuid'>{87E8D3BD-99C8-5D38-0000-002088140200}</Data><Data Name='LogonId'>0x21488</Data><Data Name='TerminalSessionId'>1</Data><Data Name='IntegrityLevel'>Medium</Data><Data Name='Hashes'>MD5=A6177D080759CF4A03EF837A38F62401,SHA256=79D1FFABDD7841D9043D4DDF1F93721BCD35D823614411FD4EAB5D2C16A86F35</Data><Data Name='ParentProcessGuid'>{87E8D3BD-9DD8-5D38-0000-00109F871D00}</Data><Data Name='ParentProcessId'>6132</Data><Data Name='ParentImage'>C:\Users\grapltest\Downloads\svchost.exe</Data><Data Name='ParentCommandLine'>.\svchost.exe</Data></EventData></Event>"#;
```

Comment:
I think this should just be in a file under a "testdata" dir or something.

Reply:
I'd rather handle that in another PR if that's alright?